### PR TITLE
feat: add user-storage to privacy snapshot and update user-storage mocks

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
note - we need to update our library to support generic paths and mock X-Sync data features

## **Description**

This fixes the flaky test for updating the privacy snapshot for user-storage API.
We also better mock the user-storage API to fail fast when testing Account-Syncing.

NOTE - we need to update our library to support generic paths, and also have mock data for testing the X-Syncing features (account syncing, network syncing, notification syncing, asset syncing).

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27292?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: privacy-snapshot.json [hunk 1]
```diff
@@ -58,5 +58,6 @@
   "oidc.api.cx.metamask.io",
   "price.api.cx.metamask.io",
   "token.api.cx.metamask.io",
-  "client-side-detection.api.cx.metamask.io"
+  "client-side-detection.api.cx.metamask.io",
+  "user-storage.api.cx.metamask.io"
```
Comment (Prithpal-Sooriya) [inline privacy-snapshot.json:62]: This API is now introduced in e2e tests as we have launched a new "Account Syncing" feature, which will sync your accounts you may have previously stored. This is done on wallet start/open post onboarding.

File: test/e2e/tests/notifications/mocks.ts [hunk 2]
```diff
@@ -34,6 +34,19 @@ export async function mockNotificationServices(server: Mockttp) {
   mockAPICall(server, await StorageMocks.getMockUserStorageGetResponse());
   mockAPICall(server, await StorageMocks.getMockUserStoragePutResponse());
 
+  // TODO - add better mock responses for other Profile Sync features
+  // (Account Sync, Network Sync, ...)
```
Comment (Prithpal-Sooriya) [inline test/e2e/tests/notifications/mocks.ts:38]: Yeah, we are currently mocking these API responses to fail fast. This ensures that the syncing features won't get in the way with existing e2e tests.

**Review by mathieuartu — APPROVED** [review]

**Review by matteoscurati — APPROVED** [review]
LGTM!

**Review by pedronfigueiredo — APPROVED** [review]

**Comment by danjm** [comment]
will this API call not occur if the basic functionalities toggle is off?

**Comment by danjm** [comment]
and when/how did the related test become flaky? how does it pass sometimes and fail other times?

**Comment by Prithpal-Sooriya** [comment]
@danjm 
> will this API call not occur if the basic functionalities toggle is off?

Yes profile sync/this API is disabled when basic functionality is turned off.

> and when/how did the related test become flaky? how does it pass sometimes and fail other times?

We started seeing this appear sometime yesterday.
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/101355/workflows/37ecc4fa-a568-4932-97b1-d90c35dfc55f/jobs/3773505
We have just released the "Account-Syncing" feature, which also is turned off if basic functionality is turned off.
The mock endpoint update was to "wildcard" all calls to this endpoint, and not just notifications.

Happy to sync or async on this.

**Pull request merged**